adapters/handlers/grpc/v1/prepare_reply.go

```diff
@@ -290,6 +293,9 @@ func extractAdditionalProps(asMap map[string]any, additionalPropsParams addition
 }
 
 func extractGroup(raw any, searchParams dto.GetParams, scheme schema.Schema, usesMarshalling bool) (*pb.GroupByResult, string, error) {
+	_, generativeSearchEnabled := searchParams.AdditionalProperties.ModuleParams["generate"]
+	_, rerankEnabled := searchParams.AdditionalProperties.ModuleParams["rerank"]
```

Comment:
move this a level up so we do not need to do this for every object?

Reply:
Ideally, this would also be the case for `extractObjectsToResults` but that would be a large refactor beyond this PR's scope